### PR TITLE
Security

### PR DESCRIPTION
**Pull request by rjkernick**

### CODE REVIEW

File: client/src/main/scala/com/all4journey/webapp/pages/LoginJsImpl.scala [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+      ),
+      div(cls:="form-group")(
+        label()(),
+        div(cls:="col-md-8")(
+          input(id :="loginButton", `type`:="button", cls:="btn btn-primary", value := "Login", onclick := {() =>
+            val userName:String = $("#userName").value().toString.trim
+            val password = $("#password").value().toString.trim
+            val loginCredentials = LoginCredentials(userName, password)
+            val pickledLogin = Pickle.intoString(loginCredentials)
+            $.ajax(js.Dynamic.literal(
```
Comment (andy-abreu) [inline client/src/main/scala/com/all4journey/webapp/pages/LoginJsImpl.scala]: This call could be replaced with AjaxHelper.doAjaxPostWithJson

File: server/src/main/scala/com/all4journey/webapp/pages/ProfilePage.scala [hunk 2]
```diff
@@ -12,6 +12,7 @@ import prickle.Unpickle
 import scala.concurrent.ExecutionContext.Implicits.global
 import scala.concurrent.{Future, Await}
 import scala.concurrent.duration.DurationInt
```
Comment (rjkernick) [inline server/src/main/scala/com/all4journey/webapp/pages/ProfilePage.scala]: Codacy Issue found: [Use brackets to group imports from the same package](https://www.codacy.com/app/rjkernick/all4journey/file/2445603283/issues/source?bid=3246460&fileBranchId=3246461#l14)

File: domain/src/main/scala/com/all4journey/domain/Tables.scala [hunk 3]
```diff
@@ -16,7 +14,7 @@ trait Tables {
   import slick.jdbc.{GetResult => GR}
```
Comment (rjkernick) [inline domain/src/main/scala/com/all4journey/domain/Tables.scala]: Codacy Issue found: [Put imports in the beginning of the package](https://www.codacy.com/app/rjkernick/all4journey/file/2445603256/issues/source?bid=3246460&fileBranchId=3246461#l14)

File: domain/src/main/scala/com/all4journey/domain/Tables.scala [hunk 4]
```diff
@@ -147,34 +145,69 @@ trait Tables {
...
+    val id: Rep[String] = column[String]("ID", O.PrimaryKey, O.Length(50,varying=true))
+    /** Database column USER_ID SqlType(VARCHAR), Length(50,true) */
+    val userId: Rep[String] = column[String]("USER_ID", O.Length(50,varying=true))
+    /** Database column TOKEN SqlType(VARCHAR), Length(50,true) */
+    val token: Rep[String] = column[String]("TOKEN", O.Length(50,varying=true))
+
+    /** Foreign key referencing User (database name TOKEN_ibfk_1) */
+    lazy val userFk = foreignKey("TOKEN_ibfk_1", userId, User)(r => r.id, onUpdate=ForeignKeyAction.NoAction, onDelete=ForeignKeyAction.Cascade)
+  }
+  /** Collection-like TableQuery object for table Token */
```
Comment (rjkernick) [inline domain/src/main/scala/com/all4journey/domain/Tables.scala]: Codacy Issue found: [Field name 'Token' does not match the regular expression '^[a-z][A-Za-z0-9]*$'](https://www.codacy.com/app/rjkernick/all4journey/file/2445603256/issues/source?bid=3246460&fileBranchId=3246461#l175)

File: domain/src/main/scala/com/all4journey/domain/Tables.scala [hunk 5]
```diff
@@ -147,34 +145,69 @@ trait Tables {
...
+    val emailAddress: Rep[String] = column[String]("EMAIL_ADDRESS", O.Length(250,varying=true))
+    /** Database column PASSWORD SqlType(VARCHAR), Length(250,true) */
+    val password: Rep[String] = column[String]("PASSWORD", O.Length(250,varying=true))
     /** Database column REGISTRATION_DATE SqlType(DATE) */
     val registrationDate: Rep[java.sql.Date] = column[java.sql.Date]("REGISTRATION_DATE")
+
+    /** Uniqueness Index over (emailAddress) (database name EMAIL_ADDRESS) */
+    val index1 = index("EMAIL_ADDRESS", emailAddress, unique=true)
   }
   /** Collection-like TableQuery object for table User */
```
Comment (rjkernick) [inline domain/src/main/scala/com/all4journey/domain/Tables.scala]: Codacy Issue found: [Field name 'User' does not match the regular expression '^[a-z][A-Za-z0-9]*$'](https://www.codacy.com/app/rjkernick/all4journey/file/2445603256/issues/source?bid=3246460&fileBranchId=3246461#l213)

File: server/src/main/scala/com/all4journey/webapp/pages/ProfilePage.scala (hunk 2 above)
Comment (rjkernick) [inline server/src/main/scala/com/all4journey/webapp/pages/ProfilePage.scala:14]: Codacy Issue found: [Use brackets to group imports from the same package](https://www.codacy.com/app/rjkernick/all4journey/file/2445603283/issues/source?bid=3246460&fileBranchId=3246461#l14)

File: domain/src/main/scala/com/all4journey/domain/Tables.scala (hunk 3 above)
Comment (rjkernick) [inline domain/src/main/scala/com/all4journey/domain/Tables.scala:14]: Codacy Issue found: [Put imports in the beginning of the package](https://www.codacy.com/app/rjkernick/all4journey/file/2445603256/issues/source?bid=3246460&fileBranchId=3246461#l14)

File: domain/src/main/scala/com/all4journey/domain/Tables.scala (hunk 4 above)
Comment (rjkernick) [inline domain/src/main/scala/com/all4journey/domain/Tables.scala:174]: Codacy Issue found: [Field name 'Token' does not match the regular expression '^[a-z][A-Za-z0-9]*$'](https://www.codacy.com/app/rjkernick/all4journey/file/2445603256/issues/source?bid=3246460&fileBranchId=3246461#l175)

File: domain/src/main/scala/com/all4journey/domain/Tables.scala (hunk 5 above)
Comment (rjkernick) [inline domain/src/main/scala/com/all4journey/domain/Tables.scala:212]: Codacy Issue found: [Field name 'User' does not match the regular expression '^[a-z][A-Za-z0-9]*$'](https://www.codacy.com/app/rjkernick/all4journey/file/2445603256/issues/source?bid=3246460&fileBranchId=3246461#l213)

**Pull request merged**